echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubMessageHandler.java

```diff
@@ -79,10 +81,18 @@ public void handleMessage(MessageDescription description,
       throw new IllegalStateException("Redis not enabled, pubsub requires redis. Please enable.");
     }
 
-    String messageKey = makeKey(description, messageId);
-    if (tryAck(messageKey, description.getAckDeadlineMillis(), acknowledger, identifier)) {
+    String completeKey = makeCompletedKey(description, messageId);
+    if (messageComplete(completeKey, description.getMessagePayload())) {
```

Comment:
I don't think there is a bound on the size of message payloads. Is there a particular reason we're storing the entire payload here rather than using hash or just an operation on the key itself?

Reply:
@jtk54 I was thinking that storing the message payload would be a good way to dedupe (if ids were the same but payload wasn't the same), but after thinking about it more it doesn't seem necessary. I'll accept that id should be unique in the retention period for popper behavior. What about storing the timestamp the message was complete, and then only checking if the key exists to see if the message was processed?

Reply:
A crc32 hash of the message would probably do just fine if you don't want to switch it up much.